radixdlt/src/integration/java/com/radixdlt/consensus/functional/synchronous/RandomDepthFirstSearchTest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus.functional.synchronous;
+
+import com.radixdlt.consensus.functional.BFTFunctionalTest;
+import java.util.Random;
+import org.junit.Test;
+
+public class RandomDepthFirstSearchTest {
```

Comment:
What makes this "depth first"?

Reply:
Maybe "depth first" isn't right term but it's basically randomly searching the markov chain n-step deep.